### PR TITLE
Custom properties for types

### PR DESCRIPTION
**Pull request by haberkornsam**
### Description ###
Added a `otherProps` map to each of the types. This catchall is very similar to the one used for leafs

### Testing ###
Added and Updated Unit Tests

### CODE REVIEW

File: src/types/__tests__/StringType-test.ts [hunk 1]
```diff
@@ -40,4 +42,14 @@ describe('String Type', () => {
 
     expect(type.serialize('foo')).to.equal('foo');
   });
```
Comment (cmain) [inline src/types/__tests__/StringType-test.ts:44]: Very minor but newlines between the tests.

File: src/model/__tests__/LeafList-test.ts [hunk 2]
```diff
@@ -67,9 +67,10 @@ describe('Leaf List Model', () => {
     const leafList = new LeafList(testModel, {} as Container);
     const expectedType = {
       baseType: {
+        otherProps: new Map(),
         type: 'union',
         types: [
-          { type: 'uint32', range: { ranges: [{ min: 0, max: 999999999 }] } },
+          { type: 'uint32', range: { ranges: [{ min: 0, max: 999999999 }] }, otherProps: new Map() },
```
Comment (BenMatase) [inline src/model/__tests__/LeafList-test.ts:73]: Is it a convention to shorten properties to props or existing key name? I usually prefer spelling everything out

Comment (haberkornsam) [reply src/model/__tests__/LeafList-test.ts:73]: `otherProps` was the convention used with leaf, leaf-list, etc. Decided it was best to keep the consistancy

Comment (cmain) [reply src/model/__tests__/LeafList-test.ts:73]: Yeah we use this same name for something similar elsewhere

**Pull request merged**